overlord/snapstate/handlers_components.go

```diff
@@ -222,6 +295,16 @@ func (m *SnapManager) doMountComponent(t *state.Task, _ *tomb.Tomb) (err error)
 	perfTimings.Save(st)
 	st.Unlock()
 
+	// TODO: test this conditional once we are installing local snaps+components
+	// if we're removing the snap file and we are mounting a component for the
+	// firs time, then we know that the component also must be coming from an
+	// emphemeral file. in that case, remove it.
+	if snapsup.RemoveSnapPath {
+		if err := os.Remove(compSetup.CompPath); err != nil {
```

Comment:
this seems not be tested

Reply:
This change shouldn't have been in this PR, it should be in the next one. My mistake.